src/docs/components/properties.md

```diff
@@ -7,219 +7,592 @@ contributors:
+})
+export class ToDoListItem {
+  @Prop() thingToDo: string;
+  
+  render() {
+    return <li>{this.thingToDo}</li>;
+  }
+}
 ```
 
-## Prop options
+> Note: Children components should not know about or reference their parent components.  This allows Stencil to
+> efficiently re-render your components. Passing a reference to a component as a prop may cause unintended side effects.
 
-The `@Prop(opts?: PropOptions)` decorator accepts an optional argument to specify certain options, such as the `mutability`, the name of the DOM attribute or if the value of the property should or shouldn't be reflected into the DOM.
+## Mutability
+
+A Prop is by default immutable from inside the component logic. Once a value is set by a user, the component cannot
+update it internally.
+
+## Types
+
+Props can be a `boolean`, `number`, `string`, or even an `Object` or `Array`.  The example below expands the 
+`todo-list-item` to add a few more props with different types.
 
 ```tsx
-export interface PropOptions {
-  attribute?: string;
-  mutable?: boolean;
-  reflect?: boolean;
+import { Component, Prop, h } from '@stencil/core';
+// `MyHttpService` is an `Object` in this example
+import { MyHttpService } from '../some/local/directory/MyHttpService';
+
+@Component({
+    tag: 'todo-list-item',
+})
+export class ToDoListItem {
+    @Prop() isComplete: boolean;
+    @Prop() timesCompletedInPast: number;
+    @Prop() thingToDo: string;
+    @Prop() myHttpService: MyHttpService;
 }
 ```
 
-### Prop mutability
+### Boolean Props
+
+A property on a Stencil component that has a type of `boolean` may be declared as:
 
-It's important to know, that a Prop is _by default_ immutable from inside the component logic. Once a value is set by a user, the component cannot update it internally.
+```tsx
+import { Component, Prop, h } from '@stencil/core';
+
+@Component({
+    tag: 'todo-list-item',
+})
+export class ToDoListItem {
+    @Prop() isComplete: boolean;
+}
+```
 
-However, it's possible to explicitly allow a Prop to be mutated from inside the component, by declaring it as **mutable**, as in the example below:
+To use this version of `todo-list-item` in HTML, we pass the string `"true"`/`"false"` to the component:
+```html
+<!-- Set isComplete to 'true' -->
+<todo-list-item is-complete="true"></todo-list-item>
+<!-- Set isComplete to 'false' -->
+<todo-list-item is-complete="false"></todo-list-item>
+```
 
+To use this version of `todo-list-item` in TSX, `true`/`false` is used, surrounded by curly braces:
 ```tsx
-import { Prop } from '@stencil/core';
+// Set isComplete to 'true'
+<todo-list-item isComplete={true}></todo-list-item>
+// Set isComplete to 'false'
+<todo-list-item isComplete={false}></todo-list-item>
+```
 
-...
-export class NameElement {
+There are a few ways in which Stencil treats props that are of type `boolean` that are worth noting:
+
+1. The value of a boolean prop will be `false` if provided the string "false" in HTML
+    ```html
```

Comment:
Do you happen to know why these code snippets are rendering weird? I don't love the inner box.

Reply:
Nor do I. Unfortunately it's an artifact of the system we use to render code snippets, and even more unfortunately I don't think there's a way around it with our current infra 😢

Reply:
Ugh that's so weird. When I changed the indentation it was fixed on a local build, so I added it here [375f115](https://github.com/ionic-team/stencil-site/pull/832/commits/375f11580f936c418175f8962dc403afd86df55f). Doesn't seem like it's fixed in the preview.

Reply:
Yeah, I couldn't get it to go away either 😆 😢 TBH I'm not too worried about it - it's not ideal, but I think we're gonna start the conversations around moving this site to Docusaurus, so hopefully it's a short lived problem

Reply:
Nvm. I forgot a newline. Fixed here [d87be29](https://github.com/ionic-team/stencil-site/pull/832/commits/d87be29ce55a48bc0884201bdaab43526f0afff9)

Reply:
Oh nice!!